pkg/catalog/export_test.go

```diff
@@ -0,0 +1,7 @@
+package catalog
+
+import "github.com/weaveworks/profiles/api/v1alpha1"
```

Comment:
Not sure why this is in a test file `_test.go`?

Reply:
Ah this is a good question.

So depending on who you ask, this is either a "valid testing pattern" or "total egregious hack". Basically by creating a `_test.go` file that is _not_ in a `_test` package, you can manipulate things like private fields as needed for testing, saving you having to make them exposed for just that purpose.

Honestly this is a convention I am still deciding whether I like or not, my gut leans towards the "egregious hack" camp, but I am happy to try it out 🤷‍♀️ .

@aclevername may also have some opinions on this.

Reply:
I get how it works, and I'm trying to work out why `.profiles` can't be exported?

In any case, this feels like it's for testing state, rather than behaviour?

Reply:
Yeah this is something that they use in the standard libraries for testing, mainly to expose private fields, [example](https://golang.org/src/math/export_test.go). Its quite handy in keeping your codes public exposure, rather than making a private field public *just* for testing you can use this functionality and keep it private. We use this elsewhere in profiles as well, for example injecting the fake httpClient https://github.com/weaveworks/profiles/blob/9734639f24ecd4dbbe9bc071235a4279ee285f60/pkg/git/export_test.go#L1-L5

Reply:
I don't think it needs to be exposed at all?

I think you can test the _behaviour_ ?

Reply:
Ah right, I didn't see what you were asking. I'll look at the test again